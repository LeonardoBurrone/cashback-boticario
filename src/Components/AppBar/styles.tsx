import { AppBar } from '@material-ui/core';
import { Theme } from '@material-ui/core/styles';
import styled from 'styled-components';

type Props = {
  theme: Theme;
};

const drawerWidth = 200;

const CBAppBar = styled(AppBar)`
  && {
    z-index: ${(props: Props) => props.theme.zIndex.drawer + 1};
    transition: ${(props: Props) =>
      props.theme.transitions.create(['width', 'margin'], {
        duration: props.theme.transitions.duration.leavingScreen,
        easing: props.theme.transitions.easing.sharp
      })};
    &.appBarShift {
      margin-left: ${drawerWidth}px;
      width: calc(100% - ${drawerWidth}px);
      transition: ${(props: Props) =>
        props.theme.transitions.create(['width', 'margin'], {
          duration: props.theme.transitions.duration.enteringScreen,
          easing: props.theme.transitions.easing.sharp
        })};
    }
  }
`;

const CBButtonDiv = styled('div')`
  && {
    display: flex;
    flex-grow: 1;
    justify-content: flex-end;
  }
`;

const LogoImg = styled('img')`
  && {
    width: 100px;
  }
`;

export { CBAppBar, CBButtonDiv, LogoImg };
