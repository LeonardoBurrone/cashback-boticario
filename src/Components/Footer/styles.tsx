import { Link } from 'react-router-dom';
import styled from 'styled-components';

const CBContainer = styled('div')`
  && {
    display: flex;
    flex-direction: column;
    margin: 30px -20px -20px -20px;
  }
`;

const CBGrid = styled('div')`
  && {
    display: flex;
    justify-content: flex-end;
    padding: 24px 24px;
  }
`;

const CBLink = styled(Link)`
  && {
    text-decoration: none;
  }
`;

export { CBContainer, CBGrid, CBLink };
