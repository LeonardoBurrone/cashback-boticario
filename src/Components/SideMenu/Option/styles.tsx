import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import styled from 'styled-components';

const OptionWrapper = styled(ListItem)`
  && {
    max-width: 200px;
    height: 70px;
    padding: 2px 10px 2px 0px;
    position: relative;
  }

  &.is-active {
    max-width: 210px;
    border-top-right-radius: 6px;
    border-bottom-right-radius: 6px;
  }
`;

const OptionIcon = styled(ListItemIcon)`
  && {
    opacity: 0.7;
    min-width: 20px;
  }
`;

const OptionText = styled(ListItemText)`
  && {
    min-width: 130px;
    white-space: normal;
    & .text-primary {
      font-size: 14px;
    }
    & .text-secondary {
      opacity: 0.7;
      font-size: 12px;
    }
  }
`;

export { OptionIcon, OptionText, OptionWrapper };
