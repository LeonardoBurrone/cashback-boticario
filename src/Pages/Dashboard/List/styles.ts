import styled from 'styled-components';

const Information = styled('div')`
  && {
    align-items: baseline;
    display: flex;
    justify-content: space-between;
  }
`;

export { Information };
