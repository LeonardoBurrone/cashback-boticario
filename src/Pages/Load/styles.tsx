import { Dialog } from '@material-ui/core';
import styled from 'styled-components';

const CBDialog = styled(Dialog)`
  & .MuiDialog-paperScrollPaper {
    align-items: center;
    padding: 30px;
  }
`;

export { CBDialog };
