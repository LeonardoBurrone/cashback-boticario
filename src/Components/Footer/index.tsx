import { Button, Grid } from '@material-ui/core';
import React from 'react';

import { CBContainer, CBGrid, CBLink } from './styles';

export type Props = {
  buttonDisabled: boolean;
  onClick: () => void;
  primaryButtonText: string;
  route?: string;
  secondaryButton?: boolean;
  secondaryButtonText?: string;
};

export const Footer: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <CBContainer>
      <CBGrid>
        <Grid container alignItems={'center'} justify={'flex-end'} spacing={3}>
          {props.secondaryButton && (
            <CBLink to={props?.route ?? '/'}>
              <Button color="primary" variant={'text'}>
                {props.secondaryButtonText}
              </Button>
            </CBLink>
          )}
          <Button color="secondary" onClick={props.onClick} variant={'contained'} disabled={props.buttonDisabled}>
            {props.primaryButtonText}
          </Button>
        </Grid>
      </CBGrid>
    </CBContainer>
  );
};

export default Footer;
