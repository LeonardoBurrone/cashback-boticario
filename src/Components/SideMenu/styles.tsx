import { Drawer } from '@material-ui/core';
import { Theme } from '@material-ui/core/styles';
import styled from 'styled-components';

type Props = {
  theme: Theme;
};

const drawerWidth = 200;

const CBDrawer = styled(Drawer)`
  && {
    width: ${drawerWidth}px;
    flex-shrink: 0;
    whitespace: 'nowrap';
    &.drawerOpen,
    &.drawerOpen .MuiDrawer-paper {
      width: ${drawerWidth}px;
      transition: ${(props: Props) =>
        props.theme.transitions.create(['width'], {
          duration: props.theme.transitions.duration.enteringScreen,
          easing: props.theme.transitions.easing.sharp
        })};
    }
    &.drawerClose,
    &.drawerClose .MuiDrawer-paper {
      transition: ${(props: Props) =>
        props.theme.transitions.create(['width'], {
          duration: props.theme.transitions.duration.leavingScreen,
          easing: props.theme.transitions.easing.sharp
        })};
      width: ${(props: Props) => props.theme.spacing(7.5)}px;
      ${(props: Props) => props.theme.breakpoints.up('sm')} {
        width: ${(props: Props) => props.theme.spacing(7.5)}px;
      }
      overflow-x: hidden;
    }
  }
`;

const OpenButtonDiv = styled('div')`
  && {
    display: flex;
    alignitems: center;
    justify-content: flex-end;
    padding: ${(props: Props) => props.theme.spacing(0, 1)};
  }
`;

export { CBDrawer, OpenButtonDiv };
