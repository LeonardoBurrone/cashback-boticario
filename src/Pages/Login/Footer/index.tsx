import { Button, Grid } from '@material-ui/core';
import React from 'react';

import { CBContainer, CBGrid, CBLink } from './styles';

type Props = {
  password: string;
  signIn: () => void;
  username: string;
};

const Footer: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <CBContainer>
      <CBGrid>
        <Grid container alignItems={'center'} justify={'flex-end'} spacing={3}>
          <CBLink to={'/sign-up'}>
            <Button variant={'text'}>Cadastrar conta</Button>
          </CBLink>
          <Button onClick={props.signIn} variant={'contained'} disabled={!props.password || !props.username}>
            Entrar
          </Button>
        </Grid>
      </CBGrid>
    </CBContainer>
  );
};

export default Footer;
