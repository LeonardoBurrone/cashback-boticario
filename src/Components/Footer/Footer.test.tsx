import { Button } from '@material-ui/core';
import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import Footer, { Props } from './';
import { CBLink } from './styles';

describe('Test component Footer', () => {
  const onClickSpy = jest.fn(() => ({}));

  const props: Props = {
    buttonDisabled: true,
    onClick: onClickSpy,
    primaryButtonText: 'Cadastre-se'
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Footer {...props} />);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('Check if component rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Call onClick when primary button is enabled', () => {
    wrapper.setProps({
      buttonDisabled: false
    });

    const primaryButton: any = wrapper.find(Button).props();
    primaryButton.onClick();

    expect(primaryButton.disabled).toBe(false);
    expect(onClickSpy).toBeCalled();
  });

  it('Check if link is set to the right route when there is a secondary button', () => {
    wrapper.setProps({
      secondaryButton: true,
      secondaryButtonText: 'Voltar'
    });

    const secondaryButton = wrapper.find(Button).at(0);

    expect(secondaryButton.exists()).toBeTruthy();
    expect(secondaryButton.prop('children')).toBe('Voltar');
    expect(wrapper.find(CBLink).props().to).toBe('/');

    wrapper.setProps({
      route: '/sign-up'
    });

    expect(wrapper.find(CBLink).props().to).toBe('/sign-up');
  });
});
