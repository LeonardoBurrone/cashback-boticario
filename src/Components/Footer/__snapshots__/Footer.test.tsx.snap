// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component Footer Check if component rendered correctly 1`] = `
<styled.div>
  <styled.div>
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      justify="flex-end"
      spacing={3}
    >
      <WithStyles(ForwardRef(Button))
        color="secondary"
        disabled={true}
        onClick={[MockFunction]}
        variant="contained"
      >
        Cadastre-se
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Grid))>
  </styled.div>
</styled.div>
`;
